src/graph.rs

```diff
@@ -183,16 +187,36 @@ impl PyGraph {
 #[pymethods]
 impl PyGraph {
     #[new]
-    #[pyo3(signature=(multigraph=true, attrs=None), text_signature = "(/, multigraph=True, attrs=None)")]
-    fn new(py: Python, multigraph: bool, attrs: Option<PyObject>) -> Self {
+    #[pyo3(signature=(multigraph=true, attrs=None, *, initial_node_count=0, initial_edge_count=0))]
+    fn new(
+        py: Python,
+        multigraph: bool,
+        attrs: Option<PyObject>,
+        initial_node_count: usize,
+        initial_edge_count: usize,
+    ) -> Self {
         PyGraph {
-            graph: StablePyGraph::<Undirected>::default(),
+            graph: StablePyGraph::<Undirected>::with_capacity(
+                initial_node_count,
+                initial_edge_count,
+            ),
             node_removed: false,
             multigraph,
             attrs: attrs.unwrap_or_else(|| py.None()),
         }
     }
 
+    fn __getnewargs_ex__<'py>(&self, py: Python<'py>) -> (Py<PyTuple>, Bound<'py, PyDict>) {
+        (
+            (self.multigraph, self.attrs.clone_ref(py)).into_py(py),
+            [
+                ("initial_node_count", self.graph.node_count()),
+                ("initial_edge_count", self.graph.edge_bound()),
```

Comment:
Silly question: why do you use `node_count` and `edge_bound` (and not say `node_bound` and `edge_bound`, or `node_count` and `edge_count`)?

Reply:
It should be `node_bound()` and `edge_bound()` because that's the size of the underlying `Vec`s in the `StableGraph` object.

Yeah that's the purpose since, it preallocates the internal `Vec`s it'll make the unpickling a bit faster.

Reply:
```suggestion
                ("initial_node_count", self.graph.node_bound()),
                ("initial_edge_count", self.graph.edge_bound()),
```

Reply:
The mystery is really why I wasn't consistent with myself on consecutive lines... Fixed in ba13b1f9.